x-pack/plugins/cases/server/client/cases/utils.ts

```diff
@@ -236,16 +240,47 @@ export const createIncident = async ({
   return { incident, comments };
 };
 
-export const getEntity = (entity: EntityInformation): string =>
-  (entity.updatedBy != null
-    ? entity.updatedBy.full_name
-      ? entity.updatedBy.full_name
-      : entity.updatedBy.username
-    : entity.createdBy != null
-    ? entity.createdBy.full_name
-      ? entity.createdBy.full_name
-      : entity.createdBy.username
-    : '') ?? '';
+export const getEntity = (
+  entity: EntityInformation,
+  userProfiles?: Map<string, UserProfile>
+): string => {
+  return (
+    getDisplayName(entity.updatedBy, userProfiles) ??
+    getDisplayName(entity.createdBy, userProfiles) ??
+    ''
+  );
+};
+
+const getDisplayName = (
+  user: User | null | undefined,
+  userProfiles?: Map<string, UserProfile>
+): string | undefined => {
+  if (user == null) {
+    return;
+  }
+
+  if (user.profile_uid != null) {
+    const updatedByProfile = userProfiles?.get(user.profile_uid);
+
+    if (updatedByProfile != null) {
+      return (
+        validOrUndefined(updatedByProfile.user.full_name) ??
+        validOrUndefined(updatedByProfile.user.username)
+      );
+    }
+  }
+
+  // TODO: if username isn't defined should we fallback to the next field (aka createdBy)? That's different than how it was before
+  return validOrUndefined(user.full_name) ?? validOrUndefined(user.username) ?? '';
+};
+
+const validOrUndefined = (value: string | undefined | null): string | undefined => {
+  if (value == null || isEmpty(value)) {
```

Comment:
This is redundant but I get a type error if I don't have the `value == null`

<img width="628" alt="image" src="https://user-images.githubusercontent.com/56361221/189760739-9a90a6ba-9fe2-4e1a-873b-4b20ebf653ab.png">